### DISCUSSION TITLE
How to decode PDF files?

### DISCUSSION THREAD

**Discussion by 1gmc1** [mono/SkiaSharp#2898] [Q&A] [answered]
I have such code:
`using var bufferStream = new MemoryStream();`
 `await stream.CopyToAsync(bufferStream);`
 `memoryStream.Position = 0;`
 `var qrCodeImage = SKBitmap.Decode(memoryStream);`
 
 It works with jpg, png and other image's formats, but returns null with pdfs. What should I do to solve this problem?

**Comment by mattleibow** [comment, marked as answer]
SkiaSharp is mainly an image generator and processor. It has a bonus feature to support generating PDFs.

It does not support reading PDFs.

Maybe try Aspose PDF.

**Reply by 1gmc1** [reply]
Ok. I see. thank u  :blush:
